PyInstaller/building/api.py

```diff
@@ -581,6 +581,11 @@ def _bootloader_file(self, exe, extension=None):
 
     def assemble(self):
         from PyInstaller.config import CONF
+
+        # On Windows, we must never create a file with a .exe suffix that we then have to (re)write to (see #6467).
+        # Any intermediate/temporary file must have an alternative suffix.
+        build_name = self.name + '.notanexecutable'
```

Comment:
Any reason why we're doing this on all OSes?

Reply:
Officially, for consistency. In reality it's so that I don't have to boot into my Windows partition to work on this.

Reply:
I guess that now CI is happy (minus the linter) I could make it conditional.

Reply:
I think I'd prefer to have it conditional, as this is Windows-specific.